File: 백준/Silver/1940. 주몽/주몽.py
import sys
input = sys.stdin.readline

N = int(input().strip())
M = int(input().strip())
num_list = list(map(int, input().split()))

# print(num_list)

num_list.sort()

left = 0
right = N - 1
answer_count = 0

while left < right:
    num_sum = num_list[left] + num_list[right]
    a = num_list[left]
    b = num_list[right]

    if num_sum < M:
        left += 1
    elif num_sum > M:
        right -= 1
    elif num_sum == M:
        answer_count += 1
        right -= 1
        left += 1

print(answer_count)

File: 백준/Silver/1940. 주몽/주몽.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(br.readLine());
        int M = Integer.parseInt(br.readLine());
        int[] array = new int[N + 1];

        StringTokenizer st = new StringTokenizer(br.readLine());

        for (int i = 1; i <= N; i++) {
            array[i] = Integer.parseInt(st.nextToken());
        }

//        System.out.println(Arrays.toString(array));
        Arrays.sort(array);

        int left = 1;
        int right = N;
        int answerCount = 0;

        while (right != left) {
            int sum = array[left] + array[right];
            if (sum < M) {
                left++;
            } else if (sum > M) {
                right--;
            } else if (sum == M) {
                answerCount++;
                left++;
            }
        }
        sb.append(answerCount);
        System.out.println(sb);
    }
}
